python/arcticdb/version_store/_store.py

```diff
@@ -1716,14 +1756,21 @@ def head(
         -------
         VersionedItem
         """
-
+        implement_read_index = "implement_read_index" in kwargs and kwargs["implement_read_index"]
+        if not implement_read_index and columns == []:
+            columns = None
         q = QueryBuilder()
         q = q._head(n)
         version_query, read_options, read_query = self._get_queries(
             symbol=symbol, as_of=as_of, date_range=None, row_range=None, columns=columns, query_builder=q, **kwargs
         )
         read_result = self._read_dataframe(symbol, version_query, read_query, read_options)
-        return self._post_process_dataframe(read_result, read_query, q)
+        index_type = read_result.norm.df.common.WhichOneof("index_type")
+        index_is_rowcount = index_type == "index" and not read_result.norm.df.common.index.is_physically_stored
+        if implement_read_index and columns == [] and index_is_rowcount:
+            return self._postprocess_df_with_only_rowcount_idx(read_result, (0, n))
```

Comment:
What if `n` is negative? (same question for tail)

Reply:
I've added tests for negative values in head and tail.